nixos/modules/profiles/installation-device.nix

```diff
@@ -105,6 +105,11 @@ with lib;
         systemdStage1Network
       ];
 
+    boot.initrd.services = {
+      lvm.enable = true;
+      swraid.enable = true;
+    };
+
```

Comment:
Why is this necessary?

Reply:
In order to provide `mdadm` and the LVM tools in installers.

Reply:
@lheckemann Not that it matters much, but shouldn't those just be added to `systemPackages`? Shouldn't need to be in initrd for the installer since the installer doesn't boot off LVM or mdraid.

Reply:
I'm not sure about LVM and swraid actually, but I know that cryptsetup fails in weird ways (opening a volume just hangs) if you only install the package and not the udev rules. The installer should have full support, not just the packages available.

Reply:
Oh sure. I guess I'm just pointing out that this does *more* than that by putting it in initrd as well, but again, it really doesn't matter much.

Reply:
Actually wait, isn't that a huge problem if lvm and swraid rules aren't in stage *2* by default anymore? We only want to make them optional in stage 1; stage 2 should have them and the ISO shouldn't have to enable them itself.

Reply:
They should not be in stage 2 either unless specifically requested by the user. The current behaviour, which places LVM in _all_ non-container system closures and activates all LVM volume groups without the user having configured anything is surprising at best and harmful at worst.

The `nixos-generate-config` changes, release notes, and `stateVersion`-based default should prevent this from breaking without notice for people who have installed their systems on LVM or mdraid.